### PR TITLE
Flight: event system refactor

### PR DESCRIPTION
**Pull request by mlyle**
Execute callbacks directly in dedicated stack.  Remove event task.  Factor system common between pipxtreme and flight boards.  

Extensive discussion of status so far in TauLabs/TauLabs#2062 .

Needs navigation flight test to merge.

### CODE REVIEW

**Comment by mlyle** [comment]
jenkins, test this please

File: flight/Modules/System/systemmod.c [hunk 1]
```diff
@@ -602,20 +672,216 @@ void vApplicationIdleHook(void)
...
+    // Create handle
+	objEntry = (PeriodicObjectList*)PIOS_malloc_no_dma(sizeof(PeriodicObjectList));
+	if (objEntry == NULL) return -1;
+	objEntry->evInfo.ev.obj = ev->obj;
+	objEntry->evInfo.ev.instId = ev->instId;
+	objEntry->evInfo.ev.event = ev->event;
+	objEntry->evInfo.cb = cb;
+	objEntry->evInfo.queue = queue;
+    objEntry->updatePeriodMs = periodMs;
+    objEntry->timeToNextUpdateMs = randomize_int(periodMs); // avoid bunching of updates
```
Comment (tracernz) [inline flight/Modules/System/systemmod.c]: indentation on these two lines :-P

File: flight/Modules/System/systemmod.c [hunk 2]
```diff
@@ -602,20 +672,216 @@ void vApplicationIdleHook(void)
...
+
+/**
+ * Handle periodic updates for all objects.
+ * \return The system time until the next update (in ms) or -1 if failed
+ */
+static uint32_t processPeriodicUpdates()
+{
+	PeriodicObjectList* objEntry;
+	int32_t timeNow;
+    int32_t offset;
```
Comment (tracernz) [inline flight/Modules/System/systemmod.c]: indentation

File: flight/Modules/System/systemmod.c [hunk 3]
```diff
@@ -602,20 +672,216 @@ void vApplicationIdleHook(void)
...
+
+	// Get lock
+	PIOS_Recursive_Mutex_Lock(mutex, PIOS_MUTEX_TIMEOUT_MAX);
+
+    // Iterate through each object and update its timer, if zero then transmit object.
```
Comment (tracernz) [inline flight/Modules/System/systemmod.c]: Weird mix of tabs and spaces from here down

File: flight/Modules/SimVisualization/simvisualization.c [hunk 4]
```diff
@@ -77,7 +77,7 @@ int32_t SimVisualizationStart(void)
 	return 0;
 }
 
-MODULE_INITCALL(VisualizationInitialize, VisualizationStart)
+MODULE_INITCALL(SimVisualizationInitialize, SimVisualizationStart)
```
Comment (tracernz) [inline flight/Modules/SimVisualization/simvisualization.c]: What's this, looks like it was always broken?

Comment (mlyle) [inline flight/Modules/SimVisualization/simvisualization.c]: Yes.  Before on the simulator the MODULE_INITCALLs weren't used and modules were "manually" invoked.  This is fixed in this PR, so it needs valid symbol names.

**Pull request merged**